File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day07SomeAssemblyRequiredTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import java.util.Map;

import static org.junit.Assert.assertEquals;

public class Day07SomeAssemblyRequiredTest {
    @Test
    public void checkReal() {
        DailyProblem<Map<String, Integer>, Map<String, Integer>> dp = new Day07SomeAssemblyRequired("/twenty_fifteen/Day07-SomeAssemblyRequired-input.txt");

        assertEquals(46065, dp.getPart1Answer().get("a").intValue());
        assertEquals(14134, dp.getPart2Answer().get("a").intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Map<String, Integer>, Map<String, Integer>> dp = new Day07SomeAssemblyRequired("/twenty_fifteen/Day07-SomeAssemblyRequired-example#1.txt");

        Map<String, Integer> part1Answer = dp.getPart1Answer();
        assertEquals(72, part1Answer.get("d").intValue());
        assertEquals(507, part1Answer.get("e").intValue());
        assertEquals(492, part1Answer.get("f").intValue());
        assertEquals(114, part1Answer.get("g").intValue());
        assertEquals(65412, part1Answer.get("h").intValue());
        assertEquals(65079, part1Answer.get("i").intValue());
        assertEquals(123, part1Answer.get("x").intValue());
        assertEquals(456, part1Answer.get("y").intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day07SomeAssemblyRequiredTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day07SomeAssemblyRequiredTest extends FlatSpec with Matchers {
    "2015 Day 7 - Input" should "calculate answers" in {
      val dp = new Day07SomeAssemblyRequired("/twenty_fifteen/Day07-SomeAssemblyRequired-input.txt")
      dp.part1Answer("a") should be(46065)
      dp.part2Answer("a") should be(14134)
    }

    "2015 Day 7 - Example #1" should "calculate answers" in {
      val dp = new Day07SomeAssemblyRequired("/twenty_fifteen/Day07-SomeAssemblyRequired-example#1.txt")
      dp.part1Answer("d") should be(72)
      dp.part1Answer("e") should be(507)
      dp.part1Answer("f") should be(492)
      dp.part1Answer("g") should be(114)
      dp.part1Answer("h") should be(65412)
      dp.part1Answer("i") should be(65079)
      dp.part1Answer("x") should be(123)
      dp.part1Answer("y") should be(456)
    }
}
